app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,69 @@
+
+class TasksController < ApplicationController
+  def index
+    @tasks = Task.all.order(:due_date , status: :desc)
+  end
+
+  def show
+    id = params[:id].to_i
+    @task = Task.find_by(id: id)
+
+    if @task.nil?
+     render :notfound, status: :not_found
+    end
+  end
+
+  def create
+    @task = Task.new(task_params) #instantiate a new book
+    @task.status = "Incomplete"
+    if @task.save # save returns true if the database insert succeeds
+      redirect_to root_path # go to the index so we can see the book in the list
+    else # save failed :(
+      render :new # show the new book form view again
+    end
+  end
+
+  def new
+    @task = Task.new(status: "Incomplete")
+  end
+
+  def edit
+    @task = Task.find(params[:id].to_i)
```

Comment:
What happens if the task isn't found? You should include error-handling code in all of these actions that look for a single task.